dali/pipeline/util/stream_pool.h

```diff
@@ -53,15 +53,14 @@ class StreamPool {
    * @brief Returns a stream from the pool. If max_size has been exceeded,
    * we hand out previously allocated streams round-robin.
    */
-  cudaStream_t GetStream() {
+  cudaStream_t GetStream(int stream_priority) {
```

Comment:
This actually can give you a stream with different priority. You could make the `stream_priority` a member of `StreamPool` and then the `GetStream` will always return streams with consistent priority.